--- a/nova/scheduler/zone.py
+++ b/nova/scheduler/zone.py
@@ -25,8 +25,11 @@
 
 from nova.scheduler import driver
 from nova import db
+from nova import flags
 
 
+FLAGS = flags.FLAGS
+
 class ZoneScheduler(driver.Scheduler):
     """Implements Scheduler as a random node selector."""
 
@@ -36,7 +39,7 @@
         """
 
         if zone is None:
-            return self.hosts_up(context, topic)
+            zone = FLAGS.default_schedule_zone
 
         services = db.service_get_all_by_topic(context, topic)
         return [service.host
--- a/nova/flags.py
+++ b/nova/flags.py
@@ -415,6 +415,9 @@
 
 DEFINE_bool('use_ipv6', False, 'use ipv6')
 
+DEFINE_string('default_schedule_zone', 'None',
+             'zone to use when user doesnt specify one')
+
 DEFINE_bool('monkey_patch', False,
               'Whether to log monkey patching')
 
