diff --git a/nova/api/ec2/cloud.py b/nova/api/ec2/cloud.py
index 5a3b952..e415a6f 100644
--- a/nova/api/ec2/cloud.py
+++ b/nova/api/ec2/cloud.py
@@ -1127,7 +1127,7 @@ class CloudController(object):
                 i['dnsNameV6'] = ip_info['fixed_ip6s'][0]
             i['privateDnsName'] = instance['hostname']
             i['privateIpAddress'] = fixed_ip
-            i['publicDnsName'] = floating_ip
+            i['publicDnsName'] = floating_ip or fixed_ip
             i['ipAddress'] = floating_ip or fixed_ip
             i['dnsName'] = i['publicDnsName'] or i['privateDnsName']
             i['keyName'] = instance['key_name']
