--- a/nova/compute/manager.py
+++ b/nova/compute/manager.py
@@ -644,7 +644,8 @@
         LOG.audit(_('instance %s: snapshotting'), instance_id,
                   context=context)
 
-        if instance_ref['power_state'] != power_state.RUNNING:
+        if (instance_ref['power_state'] != power_state.RUNNING and
+            instance_ref['power_state'] != power_state.SHUTOFF):
             state = instance_ref['power_state']
             running = power_state.RUNNING
             LOG.warn(_('trying to snapshot a non-running '
--- a/nova/virt/libvirt/connection.py
+++ b/nova/virt/libvirt/connection.py
@@ -448,7 +448,7 @@
                         '-f',
                         'qcow2',
                         '-O',
-                        'raw',
+                        'qcow2',
                         '-s',
                         snapshot_name,
                         disk_path,
@@ -464,6 +464,7 @@
 
         # Clean up
         shutil.rmtree(temp_dir)
+        snapshot_ptr.delete(0)
 
     @exception.wrap_exception()
     def reboot(self, instance, network_info):
