--- a/nova/virt/libvirt/utils.py
+++ b/nova/virt/libvirt/utils.py
@@ -225,8 +225,8 @@
     :param token: Token to require for authentication
     :param port: Port to run on
     """
-    cmd = ['%s/tools/ajaxterm/ajaxterm.py' % utils.novadir(),
-           '--command', cmd, '-t', token, '-p', port]
+    cmd = ['ajaxterm', '--command', cmd, '-t', token, '-p', port,
+           '-T', '300']
     execute(*cmd)
 
 
